### ISSUE TITLE
[Bug]  Unsupported  WITH clause 

### ISSUE THREAD

**Issue by Su1024** [apache/streampark#2348]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### Java Version

1.8

### Scala Version

2.12.x

### StreamPark Version

2.0.0

### Flink Version

flink1.13

### deploy mode

yarn-application

### What happened

Unsupported  WITH clause 

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!(您是否要贡献这个PR?)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by Su1024** [comment]
![image](https://user-images.githubusercontent.com/23012179/220284832-af45917a-bd03-4d87-827b-c5dac2dc1c86.png)

**Comment by wolfboys** [comment]
oh, this is a bug

**Comment by Su1024** [comment]
i will submit a PR to fix this bug

**Comment by wolfboys** [comment]
> i will submit a PR to fix this bug

great.

**Comment by ziqiang-wang** [comment]
> i will submit a PR to fix this bug

@wolfboys 

We have supported this syntax.
I have checked the instructions on the official website, see the screenshot for details.
According to the website, this syntax is only supported for SELECT statements, and a single WITH statement cannot be stored and executed as a single statement. It must be executed as a single SQL statement with a SELECT statement. Therefore, there are two ways to use the WITH clause:
1. Run the SELECT statement separately and view the result of the SELECT statement. However, our platform currently doesn't support running SELECT statements alone, so don't worry about that.
2. Run the INSERT statement, add one or more TIWH statements before the SELECT statement, separated by commas, and then write the SELECT statement.
I have tried the current new version of the platform, which supports the use of the WITH clause. For details, see my code example below:
```sql
create temporary table source(
  s1 string,
  s2 string
) with (
  'connector' = 'kafka',
  'topic' = 'wzq_source',
  'properties.bootstrap.servers' = '${kafka-bootstrapserver}',
  'properties.group.id' = 'test-1',
  'scan.startup.mode' = 'latest-offset',
  'scan.topic-partition-discovery.interval' = '1 h',
  'format' = 'csv'
);

create temporary table sink(
  s1 string
) with (
  'connector' = 'kafka',
  'topic' = 'wzq_sink',
  'properties.bootstrap.servers' = '${kafka-bootstrapserver}',
  'format' = 'csv'
);

insert into sink
with with1 as (
select concat(s1, '|', s2) as w1
from source
),
with2 as (
select concat(w1, '-', 'w2') as w2
from with1
)
select w2
from with2
;
```
I will post the example on the official website of the platform later.
As for your top point about writing a WITH clause, then separating it with commas and using it in the following SELECT statement, this is not supported. It is recommended that you create temporary views that will satisfy your needs if you want to use the result table of this query more than once.
